cmd/zpool/zpool_main.c

```diff
@@ -2209,34 +2209,29 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 		(void) fprintf(stderr, gettext("cannot import '%s': pool "
 		    "is formatted using an unsupported ZFS version\n"), name);
 		return (1);
-	} else if (state != POOL_STATE_EXPORTED &&
+	} else if (zfs_forceimport_required(config) && 
 	    !(flags & ZFS_IMPORT_ANY_HOST)) {
 		uint64_t hostid = 0;
-		unsigned long system_hostid = get_system_hostid();
+		char *hostname;
+		uint64_t timestamp;
+		time_t t;
 
 		(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
 		    &hostid);
```

Comment:
The cast is really not that ugly, so I dropped the extra local.  I left the "use -f" part in a separate fprintf.